dmoj/commands/validate.py

```diff
@@ -0,0 +1,179 @@
+import os
+import shlex
+import subprocess
+from itertools import groupby
+from operator import itemgetter
+from typing import List, Optional, Tuple, Union
+
+
+from dmoj import executors
+from dmoj.commands.base_command import Command
+from dmoj.contrib import contrib_modules
+from dmoj.error import CompileError, InvalidCommandException, OutputLimitExceeded
+from dmoj.judgeenv import env, get_problem_root, get_supported_problems
+from dmoj.problem import BatchedTestCase, Problem, ProblemConfig, ProblemDataManager, TestCase
+from dmoj.result import Result
+from dmoj.utils.ansi import print_ansi
+from dmoj.utils.helper_files import compile_with_auxiliary_files
+from dmoj.utils.unicode import utf8text
+
+
+all_executors = executors.executors
+
+
+class ValidateCommand(Command):
+    name = 'validate'
+    help = 'Validates input for problems.'
+
+    def _populate_parser(self) -> None:
+        self.arg_parser.add_argument('problem_ids', nargs='+', help='ids of problems to validate input')
+
+    def execute(self, line: str) -> int:
+        args = self.arg_parser.parse_args(line)
+
+        problem_ids = args.problem_ids
+        supported_problems = set(get_supported_problems())
+
+        unknown_problems = ', '.join(f"'{i}'" for i in problem_ids if i not in supported_problems)
+        if unknown_problems:
+            raise InvalidCommandException(f'unknown problem(s) {unknown_problems}')
+        total_fails = 0
+        for problem_id in problem_ids:
+            if not self.validate_problem(problem_id):
+                print_ansi(f'Problem #ansi[{problem_id}](cyan|bold) #ansi[failed](red|bold).')
+                total_fails += 1
+            else:
+                print_ansi(f'Problem #ansi[{problem_id}](cyan|bold) passed with flying colours.')
+            print()
+
+        print()
+        print('Test complete.')
+        if total_fails:
+            print_ansi(f'#ansi[A total of {total_fails} problem(s) have invalid input](red|bold)')
+        else:
+            print_ansi('#ansi[All problems validated.](green|bold)')
+
+        return total_fails
+
+    def validate_problem(self, problem_id: str) -> bool:
+        print_ansi(f'Validating problem #ansi[{problem_id}](cyan|bold)...')
+
+        problem_root = get_problem_root(problem_id)
+        if problem_root is None:
+            print_ansi(f'\t#ansi[Skipped](magenta|bold) - Problem [{problem_id}](cyan|bold) not found')
+            return True
+
+        assert problem_root is not None
```

Comment:
This shouldn't be necessary to appease mypy, since we guard with `if problem_root is None: ... return` above.